secretsource/interface.go

```diff
@@ -0,0 +1,8 @@
+package secretsource
+
+// SecretSource is the interface a secret source needs to implement
+type SecretSource interface {
+	Name() string
```

Comment:
Ideally this and whether a secret source is the default will be parsed from the configuration from the manager before they are provided to the constructor.